tests/test_rst.py

```diff
@@ -17,86 +17,66 @@
-# i = 1
-# with open(os.path.join(path, "results")) as fp:
-#     dat = fp.read().split('\n')
-# keys = dat[0:-1:7]
-# dvh = dat[4:-1:7]
-# qual = dat[5:-1:7]
-# data = {}
-# quality = {}
-# for k, v, q in zip(keys, dvh, qual):
-#     data[k] = v
-#     quality[k] = q.split()[1]
-# output = []
-# x = []
-# x2 = []
-# y = []
-# for item in datafiles:
-#     if os.path.splitext(item)[1] == ".rst":
-#         name = os.path.splitext(item)[0]
-#         rs = rst.Rst()
-#         rs.load_field(os.path.join(path, item))
-#         avg = 0
-#         i = 0
-#         value = 0
-#         # ~ for machine in rs.get_submachines():
-#         # ~ i += 1
-#         # ~ grid = machine.get_raster_grid()
-#         # ~ if len(grid) == 1 or len(grid[0]) == 1:
-#         # ~ continue
-#         # ~ dat = np.gradient(grid)
-#         # ~ norm = (dat[0]**2+dat[0]**2)**0.5
-#         # ~ value += np.sum(norm)/np.sum(norm>0)/10000
-#         # ~
-#         i = 1
-#
-#         machine = rs.get_submachines()[0]
-#         grid = machine.get_raster_grid()
-#         dat = np.gradient(grid)
-#         norm = (dat[0] ** 2 + dat[0] ** 2) ** 0.5
-#         value = np.sum(norm) / np.sum(norm > 0) / 10000
-#
-#         machine = rs.get_submachines()[1]
-#         grid = machine.get_raster_grid()
-#         dat = np.gradient(grid)
-#         norm = (dat[0] ** 2 + dat[0] ** 2) ** 0.5
-#         value += np.sum(norm) / np.sum(norm > 0) / 10000
-#
-#         machine = rs.get_submachines()[-1]
-#         grid = machine.get_raster_grid()
-#         dat = np.gradient(grid)
-#         norm = (dat[0] ** 2 + dat[0] ** 2) ** 0.5
-#         value += np.sum(norm) / np.sum(norm > 0) / 10000
-#
-#         # ~ machine = rs.get_submachines()[-2]
-#         # ~ grid = machine.get_raster_grid()
-#         # ~ dat = np.gradient(grid)
-#         # ~ norm = (dat[0]**2+dat[0]**2)**0.5
-#         # ~ value += np.sum(norm)/np.sum(norm>0)/10000
-#         # ~
-#
-#         y.append(1 - float(data[name]))
-#         x.append(value / i)
-#
-#         x2.append(float(quality[name]))
-# fig = plt.figure()
-#
-# ax = fig.add_subplot(2, 1, 1)
-# plt.plot(x, y, '.')
-#
-# ax = fig.add_subplot(2, 1, 2)
-# plt.plot(x2, y, '*')
-#
-# plt.show()
+import os
+import unittest
+import logging
+import tempfile
+# import shutil
+
+import tests.test_base
+from pytrip.raster import Rst
+import pytrip.utils.rst2sobp
+
+logger = logging.getLogger(__name__)
+
+
+class TestRst(unittest.TestCase):
+    """ Test the raster.py files
+    """
+    def setUp(self):
+        """ Prepare files for tests
+        """
+        testdir = tests.test_base.get_files()
+        self.rst000 = os.path.join(testdir, "tst003001.rst")
+        logger.info("Testing .rst from path " + self.rst000)
+
+    def test_read(self):
+        """ Check if we are able to read a simple .rst file
+        """
+        r = Rst()
+        r.read(self.rst000)
+        self.assertEqual(r.submachines, '17')
+        self.assertEqual(r.machines[0].points, 323)
```

Comment:
better would be to compare int(r.submachines) to len(r.machines), they should be equal. But there are 100x more of these comparisons which can be included. 
Feel free to do it, this is good enough to me for now.